src/Albany_Application.cpp

```diff
@@ -395,6 +395,45 @@ Application::initialSetUp(const RCP<Teuchos::ParameterList>& params)
 #endif
   }
 
+  //validate Hessian parameters
+  if(problemParams->isSublist("Hessian")) {
+    Teuchos::RCP<Teuchos::ParameterList> validHessianParams = Teuchos::rcp(new Teuchos::ParameterList("validHessianParams"));
+    validHessianParams->set<bool>("Write Hessian MatrixMarket", false);
+    validHessianParams->set<bool>("Use AD for Hessian-vector products (default)", true);
+
+    validHessianParams->sublist("Residual").set<bool>("Use AD for Hessian-vector products (default)", false);
+    validHessianParams->sublist("Residual").set<std::string>("Enable AD for Hessian-vector product contributions of", "(x,x) (p0,p0) (p0,p1) (p1,p0) (p1,p1)");
+    validHessianParams->sublist("Residual").set<std::string>("Disable AD for Hessian-vector product contributions of", "(x,p0) (x,p1) (p0,x) (p1,x)");
+
+    // We want to allow the user to add hessian settings for responses (parameters) even if those responses (parameters) are not used.
+    int maxNumResonses(3),maxNumParameters(3);
```

Comment:
Why 3?

Reply:
This is somewhat arbitrary. I don't want to choose a number that is too large, but at the same time I don't want to set it to zero, otherwise one has to delete the unused Hessian responses settings. we can chat about this.

Reply:
You could validate without recursion if num_params or num_responses is 0. And validate the sublists from inside the pfor.

Reply:
The problem is not the recursion. With the approach you suggest I cannot validate the list at the Hessian level.
Here is how a list could look like
```
Hessian:
  Residual: {}
  Response 0: 
    Parameter 0: {}
```
To validate the Hessian list I need to make a guess of how many responses are listed.

Reply:
I see, thanks.